js_runner.go

```diff
@@ -66,13 +68,39 @@ func (runner *JSRunner) Init(funcSource string) error {
 		return otto.UndefinedValue()
 	})
 
+	runner.js.Set("console", map[string]interface{}{
+		"error": func(s string) { errorMessageCallback(s)},
+		"log": func(s string) { logMessageCallback(s)},
+	})
+
 	if _, err := runner.SetFunction(funcSource); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+var logMessageCallback = defaultLogFunction
+var errorMessageCallback = defaultLogFunction
+
+var logCallbackLock sync.Mutex
+
+func (runner *JSRunner) SetErrorCallback(logFn func(string)){
```

Comment:
This could be mis-interpreted as a callback executed on an actual error, not a `console.error()` call. Could switch to something like this:

```suggestion
func (runner *JSRunner) SetConsoleErrorCallback(logFn func(string)){
```